tools/distrib/python/grpcio_tools/grpc_tools/command.py

```diff
@@ -63,3 +118,26 @@ def run(self):
         # to `self.distribution.package_dir` (and get a key error if it's not
         # there).
         build_package_protos(self.distribution.package_dir[''])
+
+
+class BuildPackageProtosStrict(setuptools.Command):
+    """Command to strictly generate project *_pb2.py modules from proto files.
+
+    The generation will abort if any of the proto files contains error.
+    """
+
+    description = 'strictly build grpc protobuf modules'
+    user_options = []
```

Comment:
I think a much simpler alternative to this PR would be to add a `strict_mode` user option to the existing `BuildPackageProtos` command that, if any compilation failure occurs, exits with a non-zero code. This satisfies the reasonable request of https://github.com/grpc/grpc/issues/17703 in a non-invasive way, and keeps the code changes here minimal.

Reply:
I had in mind using the built-in options functionality of `Command`, which our current `initialize_options()`/`finalize_options()` methods are not taking advantage of. I think this would allow the user to set an `strict_mode` option (via command-line flag, maybe programmatically) when this is invoked. I did not find great documentation around this part of `setuptools`/`distutils`, but code search turned up Tensor Flow's https://github.com/tensorflow/tensorflow/blob/3fa896d2aa9f67f88c825c3a05eff2cf01b5546a/tensorflow/tools/pip_package/setup.py#L160 which looks like it's doing something similar